CMakeLists.txt

```diff
@@ -10,6 +10,28 @@ if(POLICY CMP0076)
     cmake_policy(SET CMP0076 NEW)
 endif()
 
+add_executable(${APP_TARGET})
```

Comment:
This is here because `add_custom_command()` requires the target passed to be defined where the command is used. It did not make sense having to add `add_custom_command()`to all applications.

Reply:
This will be cleaned up via separate PR, another task to eliminate this executable and move linking there as well.

Reply:
I think we need to try and find a way to allow us to `add_executable` in the top-level application `CMakeLists.txt`. Are these custom commands going to be correct for every target? If we have to generate stuff from the tools it might be better putting them in a cmake script included by the app.

Reply:
@rwalton-arm @0xc0170 
I have just [pushed a commit](https://github.com/ARMmbed/mbed-os/pull/13378/commits/4e6495b30083a2a27aa4cb1a50ba101dd5247329) to address the above.
Please review.